packages/vue-components/src/TipBox.vue

```diff
@@ -251,86 +262,23 @@ export default {
       return {};
     },
     getBootstrapAlertStyle() {
-      switch (this.type) {
-      case 'warning':
-        return 'alert-warning';
-      case 'info':
-        return 'alert-info';
-      case 'definition':
-        return 'alert-primary';
-      case 'success':
-      case 'tip':
-        return 'alert-success';
-      case 'important':
-      case 'wrong':
-        return 'alert-danger';
-      default:
-        return 'alert-default';
-      }
+      return `alert-${this.bootStrapStyle.style}`;
     },
     getBootstrapTextStyle() {
-      switch (this.type) {
-      case 'warning':
-        return 'text-warning';
-      case 'info':
-        return 'text-info';
-      case 'definition':
-        return 'text-primary';
-      case 'success':
-      case 'tip':
-        return 'text-success';
-      case 'important':
-      case 'wrong':
-        return 'text-danger';
-      default:
-        return '';
-      }
+      return `text-${this.bootStrapStyle.style}`;
     },
     getBootstrapBorderStyle() {
-      switch (this.type) {
-      case 'warning':
-        return 'border-warning';
-      case 'info':
-        return 'border-info';
-      case 'definition':
-        return 'border-primary';
-      case 'success':
-      case 'tip':
-        return 'border-success';
-      case 'important':
-      case 'wrong':
-        return 'border-danger';
-      default:
-        return '';
-      }
+      return `border-${this.bootStrapStyle.style}`;
     },
     getFontAwesomeIconStyle() {
-      switch (this.type) {
-      case 'wrong':
-        return 'fa-times';
-      case 'warning':
-        return 'fa-exclamation';
-      case 'info':
-        return 'fa-info';
-      case 'success':
-        return 'fa-check';
-      case 'important':
-        return 'fa-flag';
-      case 'tip':
-        return 'fa-lightbulb';
-      case 'definition':
-        return 'fa-atlas';
-      default:
-        return '';
-      }
+      return this.bootStrapStyle.icon;
     },
   },
 };
 </script>
 
 <style scoped>
     .box-container {
-        width: 100%;
```

Comment:
Ok with including this here. Try to keep future PRs as atomic as possible though, it really helps those reading the commit history to trace down bugs / etc.

Reply:
Ok thanks for the feedback!